File: Problems/Algorithms/198. House Robber/house_robber.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut dp = vec![0; n+1];
        dp[1] = nums[0];
        
        for i in 2..=n {
            dp[i] = dp[i-1].max(nums[i-1]+dp[i-2]);
        }
        
        dp[n]
    }
}

File: Problems/Algorithms/198. House Robber/house_robber.py
#!/usr/bin/env python3

from typing import List


class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        
        dp = [0] * (n+1)
        dp[1] = nums[0]

        for i in range(2, n+1):
            dp[i] = max(dp[i-1], nums[i-1] + dp[i-2])
        
        return dp[n]
